### PR TITLE
Accept a single file as input

### PR DESCRIPTION
**Pull request by thompsonmj**
Addresses #13

### CODE REVIEW

File: src/sumbuddy/__main__.py [hunk 1]
```diff
@@ -9,24 +9,32 @@
...
+        file_paths = [input_path]
+        if ignore_file:
+            print("Warning: --ignore-file (-i) flag is ignored when input is a single file.")
+        if include_hidden:
+            print("Warning: --include-hidden (-H) flag is ignored when input is a single file.")
+    else:
+        try:
+            file_paths = mapper.gather_file_paths(input_path, ignore_file=ignore_file, include_hidden=include_hidden)
+        except (EmptyInputDirectoryError, NoFilesAfterFilteringError) as e:
+            sys.exit(str(e))
```
Comment (thompsonmj) [inline src/sumbuddy/__main__.py:37]: The addition of this `isfile` condition block is all that's needed for core functionality of this PR.

File: src/sumbuddy/mapper.py [hunk 2]
```diff
@@ -39,6 +39,10 @@ def gather_file_paths(self, input_directory, ignore_file=None, include_hidden=Fa
         ---------
         file_paths - List. Files in input_directory that are not ignored.
         """
+
+        if not os.path.isdir(input_directory):
+            raise NotADirectoryError(input_directory)
```
Comment (thompsonmj) [inline src/sumbuddy/mapper.py:44]: It makes more sense to me to raise an error if `gather_file_paths` receives a single file as an input than to repeat its path as an output. This doesn't interfere with use of the CLI or `get_checksums` method with a single file input.

**Review by zoeduan — COMMENTED** [review]
I run the tests, and please refer to below results:

collected 69 items                                                                                                                                                                      

tests/test_filter.py .............                                                                                                                                                [ 18%]
tests/test_getChecksums.py .........                                                                                                                                              [ 31%]
tests/test_hasher.py .................................sssss....                                                                                                                   [ 92%]
tests/test_mapper.py .....                                                                                                                                                        [100%]

============================================================================= 64 passed, 5 skipped in 0.10s

**Comment by zoeduan** [comment]
LGTM!

**Review by zoeduan — APPROVED** [review]

**Review by egrace479 — CHANGES_REQUESTED** [review]

**Review by egrace479 — APPROVED** [review]

**Pull request merged**